src/openfermion/tests/_rdm_generation.py

```diff
@@ -0,0 +1,101 @@
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Slow but simple way to generate RDMs for unit tests"""
+import os
+import numpy
+from itertools import product
+import h5py
+from openfermion.hamiltonians import MolecularData
+from openfermion.ops import FermionOperator
+from openfermion.transforms import get_sparse_operator
+from openfermion.utils import get_ground_state
+from openfermion.config import DATA_DIRECTORY
+
+
+def generate_tqdm(molecule, state):
+    """
+    Generate the 2-hole-RDM
+
+    Args:
+        molecule (MolecularData):
+        state (numpy.ndarray):
+
+    Returns:
+        tqdm (numpy.ndarray): 2-hole-RDM
+    """
+    true_tqdm = numpy.zeros_like(molecule.fci_two_rdm)
+    for p, q, r, s in product(range(molecule.n_qubits), repeat=4):
+        if (p < q and r < s and
+           p * molecule.n_qubits + q <= r * molecule.n_qubits + s):
+            fermi_op = FermionOperator(((p, 0), (q, 0), (r, 1), (s, 1)))
+            sparse_fermi_op = get_sparse_operator(fermi_op,
+                                                  n_qubits=molecule.n_qubits)
+            true_tqdm[p, q, r, s] = numpy.conj(state).T.dot(
+                sparse_fermi_op.dot(state)).real
+            true_tqdm[s, r, q, p] = true_tqdm[p, q, r, s]
+
+            true_tqdm[q, p, r, s] = -1 * true_tqdm[p, q, r, s]
+            true_tqdm[s, r, p, q] = -1 * true_tqdm[p, q, r, s]
+
+            true_tqdm[p, q, s, r] = -1 * true_tqdm[p, q, r, s]
+            true_tqdm[r, s, q, p] = -1 * true_tqdm[p, q, r, s]
+
+            true_tqdm[q, p, s, r] = true_tqdm[p, q, r, s]
+            true_tqdm[r, s, p, q] = true_tqdm[p, q, r, s]
+
+    return true_tqdm
+
+
+def generate_phdm(molecule, state):
+    """
+    Generate the particle-hole-RDM
+
+    Args:
+        molecule (MolecularData):
+        state (numpy.ndarray):
+
+    Returns:
+        phdm (numpy.ndarray): particle-hole-RDM
+    """
+    true_phdm = numpy.zeros_like(molecule.fci_two_rdm)
+    for p, q, r, s in product(range(molecule.n_qubits), repeat=4):
+        if p * molecule.n_qubits + q >= s * molecule.n_qubits + r:
+            fermi_op = FermionOperator(((p, 1), (q, 0), (r, 1), (s, 0)))
+            sparse_fermi_op = get_sparse_operator(fermi_op,
+                                                  n_qubits=molecule.n_qubits)
+            true_phdm[p, q, r, s] = numpy.conj(state).T.dot(
+                sparse_fermi_op.dot(state)).real
+            true_phdm[s, r, q, p] = true_phdm[p, q, r, s]
+
+    return true_phdm
+
+
+if __name__ == "__main__":
```

Comment:
What is this? We run tests using unittest or pytest modules. Please see other test files for an example. No modules in OpenFermion have a "main".

Reply:
It's sample code to generate marginals in a very naive way.  It's mostly an explanation of where the `tpdm_*.hdf5` files come from.  I'd be happy to remove it or move the file to the `examples/` folder in the top level directory.

Reply:
Please either add it to a demo ipython notebook in the examples folder, or remove it.

Reply:
removed